pkg_resources/__init__.py

```diff
@@ -2422,7 +2422,8 @@ def _cygwin_patch(filename):  # pragma: nocover
     return os.path.abspath(filename) if sys.platform == 'cygwin' else filename
 
 
-def _normalize_cached(filename, _cache={}):
+def _normalize_cached(filename, _cache=None):
```

Comment:
This is probably the same case of "over-clever" optimisation, isn't it?
I have the impression that the proposed change would defeat the optimisation, right?

Reply:
Same, the cache mechanism is now based on `@functools.lru_cache`.